src/tess_asteroids/movingtpf.py

```diff
@@ -487,6 +493,144 @@ def _create_threshold_mask(
 
         return aperture_mask
 
+    def _create_prf_mask(self, threshold: Union[str, float] = 0.01, **kwargs):
+        """
+        Creates an aperture mask from the PRF model.
+
+        Parameters
+        ----------
+        threshold : float or 'optimal'
+            If float, must be in the range [0,1). Only pixels where the prf model >= `threshold`
+            will be included in the aperture.
+            If 'optimal', computes optimal value for threshold.
+
+        Returns
+        -------
+        aperture_mask : ndarray
+            Boolean numpy array, where pixels inside the aperture are 'True'.
+            Shape is (ntimes, nrows, ncols).
+        """
+
+        # Create PRF model
+        if not hasattr(self, "prf_model"):
+            self._create_prf_model(**kwargs)
+
+        # Use PRF model to define aperture
+        if threshold == "optimal":
+            raise NotImplementedError(
+                "Computation of optimal PRF aperture not implemented yet."
+            )
+        elif isinstance(threshold, float) and threshold < 1 and threshold >= 0:
+            aperture_mask = self.prf_model >= threshold  # type: ignore
+        else:
+            raise ValueError(
+                f"Threshold must be either 'optimal' or a float between 0 and 1. Not '{threshold}'"
+            )
+
+        return aperture_mask
+
+    def _create_prf_model(self, time_step: float = 1.0):
```

Comment:
Maybe we want to be specific this is the target PRF model, as in the future we might want to have also PRF models for the stars when doing PSF photometry.
`_create_target_prf_model()` ?

Reply:
Is the default 1-minute step too fine for a grid for the average asteroid? For fast movers this is great, but on average asteroids move < pixel within the frame, then evaluating 30 (or more) PRFs could be doing extra work. This is irrelevant if the difference in computing time for 30 evaluations vs 5 or 10 PRFs is negligible. 
If there's a considerable difference, then maybe we want the default time step to be longer (5 min)?

Also, does `lkprf` accept fractional values for the PRF center, if so, what's the min difference (dx and dy in the center) where a PRF evaluation has different values? 

Can you test how this scales with `time_step`?

Reply:
I tested the computing time and a smaller time_step increases runtime. For example, for asteroid 1998 YT6 (sector 6, camera 1, ccd1, shape 11x11) a time_step of 1min/3min/5min had a runtime of 1.2sec/0.7sec/0.6sec. 

I have now implemented an automatic time_step, which calculates a suitable value based upon the speed of the object. A faster moving object will have a smaller time_step than a slower moving one. This provides a good compromise between runtime and resolution.